app/Repositories/ContactRepositoryCacheRedis.php

```diff
@@ -0,0 +1,38 @@
+<?php namespace App\Repositories;
+
+use App\Models\Contact;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Redis;
+
+class ContactRepositoryCacheRedis implements ContactRepository
```

Comment:
Pretty much the exact same as the cache above, except it's using the Redis facade.